dialog/internal/dialog.ts

```diff
@@ -150,7 +150,9 @@ export class Dialog extends dialogBaseClass {
   private escapePressedWithoutCancel = false;
   // This TreeWalker is used to walk through a dialog's children to find
   // focusable elements. TreeWalker is faster than `querySelectorAll('*')`.
-  private readonly treewalker = document.createTreeWalker(
+  // We check for isServer because there isn't a "document" during an SSR
+  // run.
+  private readonly treewalker = isServer ? {} as TreeWalker : document.createTreeWalker(
```

Comment:
Rather than setting it to an empty object, which could cause unexpected type failures, change the type to `TreeWalker|null`

```ts
private readonly treewalker = isServer ? null : document.createTreeWalker(...)
```

Then, check for null in `getFirstAndLastFocusableChildren()` and return early. This is more type-safe than `{} as TreeWalker`.

```ts
  private getFirstAndLastFocusableChildren() {
    if (!this.treewalker) {
      return [null, null] as const;
    }

    let firstFocusableChild: HTMLElement | null = null;
    let lastFocusableChild: HTMLElement | null = null;
```

The `as const` tells the return type to be a strict `[null, null]` tuple rather than an array of unknown length.

Reply:
Thanks for the detailed feedback, @asyncLiz.
I've added the null ternary, added the nullcheck, and removed the typecasts.
Instead of typecasting as const, I've added an explicit return type to the function, so that it can help wherever it's used.

Reply:
That sounds fine to me!